### ISSUE TITLE
SNOW-1238996: When using `deploy --replace` to change an import in a function/procedure the command isn't sent to Snowflake

### ISSUE THREAD

**Issue by sfc-gh-lfaganello** [snowflakedb/snowflake-cli#905]
### SnowCLI version

2.1.0

### Python version

3.11.5

### Platform

macOS-14.3.1-arm64-arm-64bit

### What happened

When changing an import in the definition of a function or procedure, using `deploy --replace` doesn't send the command CREATE OR REPLACE to Snowflake. We need to manually drop the function/procedure and then run `deploy`. 

### Console output

```Shell
snow snowpark deploy --connection cconnerprod3 --replace
+--------------------------------------------------------------------------------+
| object                                          | type      | status           |
|-------------------------------------------------+-----------+------------------|
| CCONNER_DB.PUBLIC.HELLO_PROCEDURE1(name string) | procedure | packages updated |
| CCONNER_DB.PUBLIC.HELLO_FUNCTION1(name string)  | function  | packages updated |
+--------------------------------------------------------------------------------+
```

### How to reproduce

```
snow connection add
snow snowpark init test
snow snowpark build
snow snowpark deploy --connection cconnerprod3
snow sql --query "call CCONNER_DB.PUBLIC.HELLO_PROCEDURE1('TESTING')" --format JSON --connection cconnerprod3
snow snowpark build
snow snowpark deploy --connection cconnerprod3 --replace
snow sql --query "create stage test" --format JSON --connection cconnerprod3
snow sql --query "file:///tmp/test.zip @test auto_compress=False" --format JSON --connection cconnerprod3
snow sql --query "put file:///tmp/test.zip @test auto_compress=False" --format JSON --connection cconnerprod3
```
Edit snowflake.yml add imports
`snow snowpark deploy --connection cconnerprod3 --replace`

After the output:
❯ snow snowpark deploy --connection cconnerprod3 --replace
```
+--------------------------------------------------------------------------------+
| object                                          | type      | status           |
|-------------------------------------------------+-----------+------------------|
| CCONNER_DB.PUBLIC.HELLO_PROCEDURE1(name string) | procedure | packages updated |
| CCONNER_DB.PUBLIC.HELLO_FUNCTION1(name string)  | function  | packages updated |
+--------------------------------------------------------------------------------+
```
When looking at the query history we can see that no command CREATE OR REPLACE was sent to Snowflake

**Comment by sfc-gh-turbaszek** [comment]
Addressed in #900 ?

**Comment by sfc-gh-pczajka** [comment]
This is addressed in #900, patch will be released in 2.1.1
